sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -348,40 +347,61 @@ public void deleteEntity(String partitionKey, String rowKey, String eTag, Durati
+     * @param partitionKey the partition key of the entity
+     * @param rowKey the row key of the entity
+     * @return the table entity
      */
-    @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PagedIterable<TableEntity> listEntities(ListEntitiesOptions options, Duration timeout) {
-        return null;
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public TableEntity getEntity(String partitionKey, String rowKey) {
+        return client.getEntity(partitionKey, rowKey).block();
     }
 
     /**
      * gets the entity which fits the given criteria
      *
      * @param partitionKey the partition key of the entity
      * @param rowKey the row key of the entity
+     * @param select a select expression using OData notation. Limits the columns on each record to just those
+     *               requested, e.g. "$select=PolicyAssignmentId, ResourceId".
      * @return the table entity
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public TableEntity getEntity(String partitionKey, String rowKey) {
-        return client.getEntity(partitionKey, rowKey).block();
+    public TableEntity getEntity(String partitionKey, String rowKey, String select) {
+        return client.getEntity(partitionKey, rowKey, select).block();
     }
 
     /**
      * gets the entity which fits the given criteria
      *
      * @param partitionKey the partition key of the entity
      * @param rowKey the row key of the entity
+     * @param select a select expression using OData notation. Limits the columns on each record to just those
+     *               requested, e.g. "$select=PolicyAssignmentId, ResourceId".
+     * @param timeout max time for query to execute before erroring out
+     * @return the table entity
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public TableEntity getEntity(String partitionKey, String rowKey, String select, Duration timeout) {
+        return getEntityWithResponse(partitionKey, rowKey, select, timeout, null).getValue();
+    }
+
+    /**
+     * gets the entity which fits the given criteria
+     *
+     * @param partitionKey the partition key of the entity
+     * @param rowKey the row key of the entity
+     * @param select a select expression using OData notation. Limits the columns on each record to just those
+     *               requested, e.g. "$select=PolicyAssignmentId, ResourceId".
+     * @param timeout max time for query to execute before erroring out
      * @param context the context of the query
      * @return a mono of the response with the table entity
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<TableEntity> getEntityWithResponse(String partitionKey, String rowKey, Context context) {
-        return client.getEntityWithResponse(partitionKey, rowKey, new QueryOptions(), context).block();
+    public Response<TableEntity> getEntityWithResponse(String partitionKey, String rowKey, String select,
```

Comment:
@alzimmermsft iirc, we use the timeout on reactor rather than the timeout to the generated rest client. (ie. `.block(timeout)`).... is this correct?

Reply:
That is the pattern that has been chosen by the Storage team

Reply:
Yeah. It would be good to know if the other language teams use the timeout parameter for the auto generated rest client or a client-side one.

Reply:
For the tables library, for each operation - the Async client does not have an overload that takes a timeout because the consumer would call `.block(timeout)` or similar to add one, whereas the Sync client does include an overload that applies a timeout. This is consistent between all operations in the Sync client except for paginated operations, because they are composed of multiple requests.